cli/command/container/stats.go

```diff
@@ -316,3 +317,31 @@ func RunStats(ctx context.Context, dockerCLI command.Cli, options *StatsOptions)
 	}
 	return err
 }
+
+// newEventHandler initializes and returns an eventHandler
+func newEventHandler() *eventHandler {
+	return &eventHandler{handlers: make(map[events.Action]func(events.Message))}
+}
+
+// eventHandler allows for registering specific events to handle.
+type eventHandler struct {
+	handlers map[events.Action]func(events.Message)
+}
+
+func (w *eventHandler) handle(action events.Action, h func(events.Message)) {
```

Comment:
Very minor nits here:
- `w` as the eventHandler name? Could we use `eh` like we do above?
- For the naming i'd suggest something along the lines of `setHandler`/`setActionHandler` etc since we don't actually do any handling in here. WDYT?